# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix Romanian locale
# DP: Dpatch author: Eddy Petrisor  <eddy.petrisor@gmail.com>
# DP: Patch author: Eddy Petrisor  <eddy.petrisor@gmail.com>
# DP: Upstream status: BZ2125
# DP: Date: 2006-02-24

2006-01-07  Eddy Petrisor  <eddy.petrisor@gmail.com>

	* locales/ro_RO: Correct the sorting order of the letters a 
	circumflex and a with breve according to the Romanian alphabet.
	* locales/ro_RO: Do not use capital A with breve within day names
	* locales/ro_RO: Use Romanian post-92 writing rules within day
	* locales/ro_RO: After denomination starting with the 1st of July 2005,
	int'l currency symbol is RON (1 RON = 10000 ROL);
	see http://publications.eu.int/code/en/en-5000700.htm; 
	* locales/ro_RO: groupping sign for thousands is "."; group of 3
	* locales/ro_RO: short date format is %d.%m.%Y for RO
	* locales/ro_RO: placed year before time in date_fmt
	* locales/ro_RO: replaced %Z with %z in date formats because %Z is not
	used nor widely known in Romania, and Romania uses daylight saving and
	the difference is more obvious this way
	* locales/ro_RO: changed abday for Saturday as i> looks bad and is
	incorrect according to post-92 rules
	* locales/ro_RO: do not capitalize months and days as it is not correct
	in Romanian
	* locales/ro_RO: A4 is the prefered paper type; metric system is used
	(removed FIXMEs)
	* locales/ro_RO: added country_name, country_car, lang_name and lang_ab
	* locales/ro_RO: added name_mr, name_mrs, name_miss (name_ms omitted as
	there is no such proper form in Romanian)
	* locales/ro_RO: added explanation related to the cedilla/comma issue and
	the reson why the transliteration is a good idea
	* locales/ro_RO: changed default encoding to UTF-8 - this is the only 
	encoding that supports all Romanian specific symbols (see encoding table
	in Debian BTS, #119528 and the corresponding comments in #347173)
	* locales/ro_RO: Corrected the name format (salutation abbreviation was
	omitted)
	* locales/ro_RO: Corrected postal_fmt (See address examples from
	Romanian Ministries' sites in Debian BTS #347173)
	* locales/ro_RO: first_weekday and first_workday are both Monday
	* locales/ro_RO: added terminology and bibliographic codes for RO
	reference: http://www.loc.gov/standards/iso639-2/langcodes.html#qr
	* locales/ro_RO: added isbn code
	* locales/ro_RO: added postal code - RO (not 100% sure)

--- localedata/locales/ro_RO.orig	Thu Feb 23 14:19:22 2006
+++ localedata/locales/ro_RO	Fri Feb 24 00:14:36 2006
@@ -15,9 +15,13 @@
 % Date: 1996-10-15
 % Application: general
 % Users: general
-% Charset: ISO-8859-2
+% Charset: UTF-8
 % Distribution and use is free, also
 % for commercial purposes.
+%
+% Corrections by:
+%  2006-01-07
+%  Eddy Petrisor <eddy.petrisor@gmail.com>
 
 LC_IDENTIFICATION
 title      "Romanian locale for Romania"
@@ -51,10 +55,11 @@
 % Copy the template from ISO/IEC 14651
 copy "iso14651_t1"
 
-% a a> a( b c d e f g h i i> j k l m n o p q r s s, t t, u v w x y z
+% The modern Romanian alphabet has 31 letters, ordered as below:
+% a a( a> b c d e f g h i i> j k l m n o p q r s s, t t, u v w x y z
 
-collating-symbol <a/>>
 collating-symbol <a(>
+collating-symbol <a/>>
 collating-symbol <i/>>
 collating-symbol <s,>
 collating-symbol <t,>
@@ -64,8 +69,8 @@
 <BRE>
 
 reorder-after <a>
-<a/>>
 <a(>
+<a/>>
 reorder-after <i>
 <i/>>
 reorder-after <s>
@@ -74,11 +79,11 @@
 <t,>
 
 reorder-after <U0061>
-<U00E2> <a/>>;<CIR>;<MIN>;IGNORE
 <U0103> <a(>;<BRE>;<MIN>;IGNORE
+<U00E2> <a/>>;<CIR>;<MIN>;IGNORE
 reorder-after <U0041>
-<U00C2> <a/>>;<CIR>;<CAP>;IGNORE
 <U0102> <a(>;<BRE>;<CAP>;IGNORE
+<U00C2> <a/>>;<CIR>;<CAP>;IGNORE
 
 reorder-after <U0069>
 <U00EE> <i/>>;<CIR>;<MIN>;IGNORE
@@ -103,9 +108,22 @@
 
 END LC_COLLATE
 
+
+
 LC_CTYPE
 copy "i18n"
 
+%
+% Transliteration is _currently_ a good idea as most fonts/applications do not
+% have the correct diacritics for "t comma bellow" (U021A, U021B) and 
+% "s comma bellow" (U0218, U0219). Thus we currently use and compromise with
+% "t cedilla" (U0162, U0163) - which, actually, does not belong to any 
+% language - and "s cedilla" (U015E, U015F) - which is _not_ correct for
+% Romanian, but is correct for Turkish.
+%
+% This issue should be fixed in the future, when enough font and application
+% support will be provided in current operating systems.
+%
 translit_start
 include "translit_combining";""
 
@@ -124,19 +142,31 @@
 translit_end
 END LC_CTYPE
 
+
+
 LC_MESSAGES
 yesexpr "<U005E><U005B><U0044><U0064><U0059><U0079><U005D><U002E><U002A>"
 noexpr  "<U005E><U005B><U006E><U004E><U005D><U002E><U002A>"
 END LC_MESSAGES
 
+
 LC_MONETARY
-int_curr_symbol           "<U0052><U004F><U004C><U0020>"
+% Since the 1st of July 2005, the new currency is the ROmanian New leu
+% the symbol is RON and 1 RON = 10000 ROL (old ROmanian Leu)...
+int_curr_symbol           "<U0052><U004F><U004E><U0020>"
+% but representing counted coins is made in the same way as before.
 currency_symbol           "<U004C><U0065><U0069>"
+% decimal point is comma
 mon_decimal_point         "<U002C>"
+% the separation of digits is made in groups of three
+% and the separator between the groups is the full stop (dot)
 mon_thousands_sep         "<U002E>"
 mon_grouping              3;3
+% the positive sign is not usualy represented
 positive_sign             ""
+% while the negative one is
 negative_sign             "<U002D>"
+% usualy one will display only two digits after the decimal point
 int_frac_digits           2
 frac_digits               2
 p_cs_precedes             1
@@ -147,80 +177,213 @@
 n_sign_posn               1
 END LC_MONETARY
 
+
 LC_NUMERIC
+% same rules as for monetary apply for regular figures:
+% decimal point is comma
 decimal_point             "<U002C>"
-thousands_sep             ""
-grouping                  0;0
+% and the separator between groups is full stop
+thousands_sep             "<U002E>"
+% the separation of figures is made in groups of three
+grouping                  3;3
 END LC_NUMERIC
 
+
 LC_TIME
+% abbreviated day names are (starting from Monday): Lu Ma Mi Jo Vi Sb Du
 abday   "<U0044><U0075>";"<U004C><U0075>";"<U004D><U0061>";"<U004D><U0069>";/
-        "<U004A><U006F>";"<U0056><U0069>";"<U0053><U00EE>"
-day     "<U0044><U0075><U006D><U0069><U006E><U0069><U0063><U0102>";/
-        "<U004C><U0075><U006E><U0069>";/
-        "<U004D><U0061><U0072><U0163><U0069>";/
-        "<U004D><U0069><U0065><U0072><U0063><U0075><U0072><U0069>";/
-        "<U004A><U006F><U0069>";/
-        "<U0056><U0069><U006E><U0065><U0072><U0069>";/
-        "<U0053><U00EE><U006D><U0062><U0102><U0074><U0102>"
+        "<U004A><U006F>";"<U0056><U0069>";"<U0053><U0062>"
+%
+% Because in 1992 the Romanian Academy stated that within words the
+% writing of the letter i> (i circumflex - U00EE) should be replaced
+% by a> (acircumflex - U00E2), we use the post92 writing.
+% Note: The actual rule is more complex, but the case which applies for 
+% Satruday is covered by the exposed explanation above.
+%
+% Day names are not capitalized:
+% duminic<abreve>, luni, mar<tcomma>i, 
+% miercuri, joi, vineri,
+% s<acircumflex>mb<abreve>t<abreve>
+%
+% Note that Tuesday Romanian day should be using t comma, not t cedilla,
+% but due to the low availability of the glyph (U021B) in fonts we use
+% the "so called t cedilla"
+day     "<U0064><U0075><U006D><U0069><U006E><U0069><U0063><U0103>";/
+        "<U006C><U0075><U006E><U0069>";/
+        "<U006D><U0061><U0072><U0163><U0069>";/
+        "<U006D><U0069><U0065><U0072><U0063><U0075><U0072><U0069>";/
+        "<U006A><U006F><U0069>";/
+        "<U0076><U0069><U006E><U0065><U0072><U0069>";/
+        "<U0073><U00E2><U006D><U0062><U0103><U0074><U0103>"
+%
+% All month abbreviations are composed of the first three letters of the full
+% Romanian names of the months, except for November, whose abbreviation comes
+% from the Latin month name, thus the abbreviation is "nov"
+%
+% Month names are not capitalized:
+% ian feb mar apr mai iun iul aug sep oct nov dec
 abmon   "<U0069><U0061><U006E>";"<U0066><U0065><U0062>";/
         "<U006D><U0061><U0072>";"<U0061><U0070><U0072>";/
         "<U006D><U0061><U0069>";"<U0069><U0075><U006E>";/
         "<U0069><U0075><U006C>";"<U0061><U0075><U0067>";/
         "<U0073><U0065><U0070>";"<U006F><U0063><U0074>";/
         "<U006E><U006F><U0076>";"<U0064><U0065><U0063>"
-mon     "<U0049><U0061><U006E><U0075><U0061><U0072><U0069><U0065>";/
-        "<U0046><U0065><U0062><U0072><U0075><U0061><U0072><U0069><U0065>";/
-        "<U004D><U0061><U0072><U0074><U0069><U0065>";/
-        "<U0041><U0070><U0072><U0069><U006C><U0069><U0065>";/
-        "<U004D><U0061><U0069>";/
-        "<U0049><U0075><U006E><U0069><U0065>";/
-        "<U0049><U0075><U006C><U0069><U0065>";/
-        "<U0041><U0075><U0067><U0075><U0073><U0074>";/
-        "<U0053><U0065><U0070><U0074><U0065><U006D><U0062><U0072><U0069><U0065>";/
-        "<U004F><U0063><U0074><U006F><U006D><U0062><U0072><U0069><U0065>";/
-        "<U004E><U006F><U0069><U0065><U006D><U0062><U0072><U0069><U0065>";/
-        "<U0044><U0065><U0063><U0065><U006D><U0062><U0072><U0069><U0065>"
-d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
-d_fmt   "<U0025><U0059><U002D><U0025><U006D><U002D><U0025><U0064>"
+%
+% Month names are _not_ capitalized in Romanian:
+% ianuarie februarie martie
+% aprilie mai iunie
+% iulie august septembrie
+% octombrie noiembrie decembrie
+mon     "<U0069><U0061><U006E><U0075><U0061><U0072><U0069><U0065>";/
+        "<U0066><U0065><U0062><U0072><U0075><U0061><U0072><U0069><U0065>";/
+        "<U006D><U0061><U0072><U0074><U0069><U0065>";/
+        "<U0061><U0070><U0072><U0069><U006C><U0069><U0065>";/
+        "<U006D><U0061><U0069>";/
+        "<U0069><U0075><U006E><U0069><U0065>";/
+        "<U0069><U0075><U006C><U0069><U0065>";/
+        "<U0061><U0075><U0067><U0075><U0073><U0074>";/
+        "<U0073><U0065><U0070><U0074><U0065><U006D><U0062><U0072><U0069><U0065>";/
+        "<U006F><U0063><U0074><U006F><U006D><U0062><U0072><U0069><U0065>";/
+        "<U006E><U006F><U0069><U0065><U006D><U0062><U0072><U0069><U0065>";/
+        "<U0064><U0065><U0063><U0065><U006D><U0062><U0072><U0069><U0065>"
+% Appropriate date and time representation (%c)
+% Romania uses daylight saving and the names of the time zones are 
+% not widely known nor used
+%	"%a %d %b %Y %T %z"
+d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020>/
+<U0025><U0059><U0020><U0025><U0054><U0020><U0025><U007A>"
+%
+% Appropriate date representation (%x)
+%	"%d.%m.%Y"
+d_fmt   "<U0025><U0064><U002E><U0025><U006D><U002E><U0025><U0059>"
+% 24 hour format is used,
 t_fmt   "<U0025><U0054>"
+%  thus no am/pm markers are defined/used
 am_pm   "";""
+% also am/pm hour display format is not used
 t_fmt_ampm ""
-date_fmt       "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
-<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
-<U0025><U005A><U0020><U0025><U0059>"
+% %A %-e %B %Y, %H:%M:%S %z
+date_fmt       "<U0025><U0041><U0020><U0025><U002D><U0065><U0020><U0025><U0042>/
+<U0020><U0025><U0059><U002C><U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U007A>"
+% First week day is Monday
+first_weekday 2
+% First work day is Monday
+first_workday 2
 END LC_TIME
 
+
+% Romania uses A4 paper format
 LC_PAPER
-% FIXME
 height   297
-% FIXME
 width    210
 END LC_PAPER
 
+
 LC_TELEPHONE
 tel_int_fmt    "<U002B><U0025><U0063><U0020><U0025><U0061><U0020><U0025>/
 <U006C>"
 int_prefix     "<U0034><U0030>"
 END LC_TELEPHONE
 
+
+% Romania uses the metric system
 LC_MEASUREMENT
-% FIXME
 measurement    1
 END LC_MEASUREMENT
 
+
 LC_NAME
-name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
-<U0025><U006D><U0025><U0074><U0025><U0066>"
+% Format of a name, taken from the glibc locale for Ukrainian.
+% %f Family names.
+% %F Family names in uppercase. 
+% %g First given name.
+% %G First given initial.
+% %l First given name with Latin letters.
+% %o Other shorter name.
+% %m Additional given names.
+% %M Initials for additional given names.
+% %p Profession.
+% %s Salutation, such as "Doctor"
+% %S Abbreviated salutation, such as "Mr." or "Dr."
+% %d Salutation, using the FDCC-sets conventions
+%    1 for the name_gen
+%    2 for name_mr
+%    3 for name_mrs
+%    4 for name_miss
+%    5 for name_ms
+% %t If the preceding field descriptor resulted in an empty string,
+%    then the empty string, else a <space>.
+%
+% FIXME: There are two prefered ways:
+% %d%t%s%t%g%t%f and %d%t%s%t%f%t%g%t%m
+% The current tendency is to use the first form, while the second is the most
+% popular form, it is widespread and widely known.
+% Now we stick to the traditional/old/second form: %d%t%s%t%f%t%g%t%m
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0073><U0025>/
+<U0074><U0025><U0066><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D>"
+% Dl.
+name_mr     "<U0044><U006C><U002E>"
+% D-na.
+name_mrs    "<U0044><U002D><U006E><U0061><U002E>"
+% D-ra.
+name_miss     "<U0044><U002D><U0072><U0061><U002E>"
 END LC_NAME
 
 LC_ADDRESS
-postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
-<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
-<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
-<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
-<U004E><U0025><U0063><U0025><U004E>"
+% Format of an address. The fields have the following meaning
+% (inspired originaly by glibc locale for Ukranian):
+
+% %n Person's name, possibly constructed with the LC_NAME
+%    "name_fmt" keyword. (it appears to be invalid)
+% %a Care of person, or organization. 
+% %f Firm name. 
+% %d Department name. 
+% %b Building name. 
+% %s Street or block (eg. Japanese) name. 
+% %h House number or designation. 
+% %N Insert an <end-of-line> if the previous descriptor
+%    value was not an empty string; otherwise ignore.
+% %t Insert a <space> if the previous descriptor value
+%    was not an empty string; otherwise ignore.
+% %r Room number, door designation. 
+% %e Floor number. 
+% %C Country designation, from the <country_post> keyword. 
+% %l Local township within town or city. (it appears to be invalid)
+% %z Zip number, postal code. 
+% %T Town, city. 
+% %S State, province, or prefecture. 
+% %c Country, as taken from data record.
+%
+% FIXME: it appears that %n and %l are not working; correct form is:
+% %n%N%f%N%a%d%N%s%t%h%N%b%t%e%t%r%N%l%z%t%T%N%S%t%c%N
+% implemented form:
+% %f%N%a%d%N%s%t%h%N%b%t%e%t%r%N%z%t%T%N%S%t%c%N
+postal_fmt    "<U0025><U0066>/
+<U0025><U004E><U0025><U0061><U0025><U0064><U0025><U004E>/
+<U0025><U0073><U0025><U0074><U0025><U0068><U0025><U004E>/
+<U0025><U0062><U0025><U0074><U0025><U0065><U0025><U0074>/
+<U0025><U0072><U0025><U004E><U0025><U007A>/
+<U0025><U0074><U0025><U0054><U0025><U004E><U0025><U0053>/
+<U0025><U0074><U0025><U0063><U0025><U004E>"
+% Country names are capitalized: Roma>nia
+country_name "<U0052><U006F><U006D><U00E2><U006E><U0069><U0061>"
 country_ab2 "<U0052><U004F>"
 country_ab3 "<U0052><U004F><U0055>"
 country_num 642
+country_car "<U0052><U004F>"
+% ISBN code is 973
+% see: http://homepages.cwi.nl/~dik/english/codes/isbn.html
+% and other sources
+country_isbn 973
+% FIXME: is it really RO?
+country_post "<U0052><U004F>"
+% language names are not capitalized in Romanian ( roma>na( )
+lang_name "<U0072><U006F><U006D><U00E2><U006E><U0103>"
+lang_ab "<U0072><U006F>"
+% set the terminology code to "ron" as defined in iso639-2
+lang_term "<U0072><U006F><U006E>"
+% set the bibliographic code to "rum" as defined in iso639-2
+lang_lib  "<U0072><U0075><U006D>"
 END LC_ADDRESS
